sdk/ai/Azure.AI.Agents.Persistent/src/Telemetry/ScopeDisposingPageable.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Azure.AI.Agents.Persistent.Telemetry
+{
+    internal class ScopeDisposingPageable<T> : Pageable<T>
```

Comment:
pageable is not disposable and there is nothing in our docs that recommends disposing it.

Users are free to walk away from pageable at any moment and aren't expected to read it to the end. I.e. it's not possible to instrument listing-all. We can only instrument individual page retrievals.